tools/unused_modules.py

```diff
@@ -0,0 +1,75 @@
+import os
+import re
+
+rolesPath = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'roles'))
+modulesPath = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'module'))
+
+basePath = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'module')) + os.sep
+
+existingModules = {}
+missingModules = {}
+commentedModules = {}
+usedModules = []
+unusedButCommentedModules = []
+
+# a list of folders that could be within a module
+skip_list = ['src', 'data', 'config', 'tests', 'report']
+
+# find all the modules that are available
+for root, dirs, files in os.walk(modulesPath):
+	if len(files) != 0:
```

Comment:
Should be indenting with spaces not tabs. You should open the NUbots project in sublime and it should ensure all this stuff for you.